zerver/webhooks/rundeck/tests.py

```diff
@@ -0,0 +1,48 @@
+from zerver.lib.test_classes import WebhookTestCase
+
+
+class RundeckHookTests(WebhookTestCase):
+    STREAM_NAME = "Rundeck"
+    TOPIC_NAME = "Global Log Filter Usage"
+    URL_TEMPLATE = "/api/v1/external/rundeck?&api_key={api_key}&stream={stream}"
+    WEBHOOK_DIR_NAME = "rundeck"
+
+    def test_start_message(self) -> None:
+        expected_message = "Job Execution [STARTED](https://rundeck.com/project/myproject/execution/follow/12) :running:"
+
+        self.check_webhook(
+            "start",
+            RundeckHookTests.TOPIC_NAME,
+            expected_message,
+            content_type="application/x-www-form-urlencoded",
+        )
+
+    def test_success_message(self) -> None:
+        expected_message = "Job Execution [SUCCEEDED](https://rundeck.com/project/myproject/execution/follow/12) :check:"
+
+        self.check_webhook(
+            "success",
+            RundeckHookTests.TOPIC_NAME,
+            expected_message,
+            content_type="application/x-www-form-urlencoded",
+        )
+
+    def test_failure_message(self) -> None:
+        expected_message = "Job Execution [FAILED](https://rundeck.com/project/myproject/execution/follow/13) :cross_mark:"
+
+        self.check_webhook(
+            "failure",
+            RundeckHookTests.TOPIC_NAME,
+            expected_message,
+            content_type="application/x-www-form-urlencoded",
+        )
+
+    def test_duration_message(self) -> None:
+        expected_message = "Job Execution [RUNNING LONG](https://rundeck.com/project/myproject/execution/follow/13) :time_ticking:"
```

Comment:
Can you rework the formatting to use ? None of our other integrations for job running platforms use all caps text or weird mixed case or a non-sentence. `zerver/webhooks/codeship/view.py` `zerver/webhooks/gocd/view.py` may be useful examples. But I think we want a sentence like `[Job 13](url) succeeded.`, where the URL is on the label for it, etc.

That said, there's also other information that almost certainly should be in this notice.
```
    "job": {
      "id": "a0296d93-4b10-48d7-8b7d-86ad3f603b85",
      "name": "Global Log Filter Usage",
      "group": "Basic Examples/Basic Workflows",
      "project": "MyProject",
      "description": "Global Log Filter basic example. \n\nMore information: https://docs.rundeck.com/docs/manual/log-filters/#log-filters",
      "href": "https://rundeck.com/api/1/job/a0296d93-4b10-48d7-8b7d-86ad3f603b85",
      "permalink": "https://rundeck.com/project/myproject/job/show/a0296d93-4b10-48d7-8b7d-86ad3f603b85"
    }

```

So probably something like:

`[{job_name}]({job_ref}) execution [{execution_id}]({execution_href}) for {project} succeeded :check:.`

Reply:
I have updated the fixtures - the structure is changed a bit now, but it doesn't affect the integration. Also changed the formatting as suggested above and improved the messages' wording.